***Из колоды в 52 карты извлекаются случайным образом 4 карты.***
---
- a) Найти вероятность того, что все карты – крести. 

52 карты - стандартная колода, в которой находятся 4 масти - трефы, бубны, пики, червы. 

У каждой из масти есть 13 достоинств, начиная от 2, заканчивая тузом. 

Вероятность, что все карты крести(трефы) - это сумма вероятностей каждой отдельно взятой карты последовательно.


Вероятность, что первая карта окажется с мастью крести = 13/52.

Вероятность, что вторая взятая карта после первой тоже будет крести = 12/51, так как:
 - Мы уже взяли одну карту крести, соответственно удаляем ее из списка 
 - И так как на одну карту стало меньше - уменьшаем и общее кол-во карт.

In [2]:
one = 13/52
two = 12/51
three = 11/50
four = 10/49

one, two, three, four

(0.25, 0.23529411764705882, 0.22, 0.20408163265306123)

Умножая эти вероятности, мы получим полную вероятность того, что все карты - крести.

In [3]:
one * two * three * four

0.0026410564225690276

Ответ: с вероятностью в 0.26% мы получим 4 карты крести подряд

- б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

In [4]:
from math import factorial

Так как у нас всего 4 масти, значит всего 4 туза.

1) Определяем общее количество комбинаций

In [8]:
full = factorial(52) / (factorial(4) * factorial(52 - 4))
full

270725.0

2) Определяем количество комбинаций без туза

In [11]:
no_tuz = factorial(48) / (factorial(4) * factorial(48 - 4))
no_tuz

194580.0

Соответственно вычитаем из 100% вероятность того, что не будет туза вообще.

In [13]:
1 - (no_tuz / full)

0.2812632745405854

Ответ: С вероятностью в 28.1% мы получим хотя бы 1 туз в рукаве.